src/python/pants/engine/objects.py

```diff
@@ -167,3 +169,41 @@ def __iter__(self) -> Iterator[_C]:
 
   def __bool__(self) -> bool:
     return bool(self.dependencies)
+
+
+@sentinel_attribute('_is_union')
```

Comment:
Would we be able to drop the argument by referring instead to the symbol itself at a fixed field name for the check?

I'm imagining something like:
```python
@sentinel_attribute
def union(cls):
  ...


def is_instance(self, obj: type) -> bool:
  return obj._sentinel_attribute_type == self.__class__._sentinel_attribute_type
```
or something along those lines (depending on exactly what `self`/`cls`/metavariables exist in the context where `is_instance` is actually defined)

Reply:
Thank you!!! Implemented!!!